WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.java

```diff
@@ -78,14 +90,126 @@ public void onResume() {
     }
 
     @Override
+    protected void onPause() {
+        super.onPause();
+        if (mSearchMenuItem != null) {
+            String tempQuery = mQuery;
+            MenuItemCompat.collapseActionView(mSearchMenuItem);
+            mQuery = tempQuery;
+        }
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.posts_list, menu);
+
+        mSearchMenuItem = menu.findItem(R.id.search_posts_list);
+        MenuItemCompat.setOnActionExpandListener(mSearchMenuItem, this);
+
+        mSearchView = (SearchView) mSearchMenuItem.getActionView();
+        mSearchView.setOnQueryTextListener(this);
+
+        // open search bar if we were searching for something before
+        if (!TextUtils.isEmpty(mQuery)) {
+            String tempQuery = mQuery; //temporary hold onto query
+            MenuItemCompat.expandActionView(mSearchMenuItem); //this will reset mQuery
+            onQueryTextSubmit(tempQuery);
+            mSearchView.setQuery(mQuery, true);
+        }
+
+        return super.onCreateOptionsMenu(menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                onBackPressed();
+                return true;
+            case R.id.search_posts_list:
+                mSearchMenuItem = item;
+                mSearchView = (SearchView) item.getActionView();
+                mSearchView.setOnQueryTextListener(this);
+
+                if (!NetworkUtils.checkConnection(this)) {
```

Comment:
> If I understand this check correctly, it's trying to prevent that

Correct, maybe it's better to just disable it. I didn't want to add a network event handler but it might be the best option here.